pipelines/pipelines/build-image-webapp.yaml

```diff
@@ -0,0 +1,97 @@
+#
+# Copyright contributors to the Galasa project 
+# 
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: build-image-webapp
+  namespace: galasa-pipelines
+spec:
+  workspaces:
+  - name: git-workspace
+  params:
+  - name: distBranch
+    type: string
+  - name: refspec
+    type: string
+  - name: noPush
+    type: string
+    default: "--no-push"
+  - name: imageName
+    type: string
+    default: nopushimage:latest
+  - name: context
+    value: simplatform/galasa-simplatform-application/galasa-simplatform-webapp   
+
+  tasks:
+  - name: clean-simplatform
+    taskRef: 
+      name: git-clean
+    params:
+    - name: subdirectory
+      value: simplatform
+    workspaces:
+     - name: git-workspace
+       workspace: git-workspace
+
+  - name: clone-simplatform
+    taskRef: 
+      name: git-clone
+    runAfter:
+    - clean-simplatform
+    params:
+    - name: url
+      value: https://github.com/galasa-dev/simplatform
+    - name: revision
+      value: $(params.distBranch)
+    - name: refspec
+      value: $(params.refspec)
+    - name: depth
+      value: "99999999"
+    - name: subdirectory
+      value: simplatform
+    workspaces:
+     - name: output
+       workspace: git-workspace
+
+  - name: maven-build-simplatform
+    taskRef:
+      name: maven-build
+    runAfter:
+    - clone-simplatform    
+    params:
+    - name: context
+      value: simplatform/galasa-simplatform-application
+    - name: settingsPath
+      value: simplatform/settings.xml
+    - name: dockerRepo
+      value: harbor-cicsk8s.hursley.ibm.com 
+    - name: build-args
+      value:  
+        - "-Dgpg.skip=true"
+        - "-Dgalasa.release.repo=file:/workspace/git/simplatfom/docker/repo/"
+        - "-Dgalasa.source.repo=https://artifactory-cicsk8s.hursley.ibm.com/artifactory/maven-proxy/"
+        - "-Dgalasa.central.repo=https://artifactory-cicsk8s.hursley.ibm.com/artifactory/maven-proxy/" 
```

Comment:
Is this an internal or external build?

Reply:
I'm not too sure. I think it could also be a mvn install rather than deploy here. these arguments are for the settings.xml but I'm not sure if it is needed and if they're then I'm not too sure what the values should be - I got these from the default values outputted when experimenting with the other Tekton pipelines

Reply:
I have changed these values in the third commit